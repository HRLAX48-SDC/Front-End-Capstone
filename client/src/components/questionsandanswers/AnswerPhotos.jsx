import React, {useState} from 'react';
import { Image, Modal, closeButton } from 'react-bootstrap';

const AnswerPhotos = (props) => {
  const [isEnlarge, setLarge] = useState({
    showed: false,
    index: 0
  });
  console.log(props.photos);
  return (
    <div className='a_photos'>
      {props.photos.map((photo, i) =>
        <div key={i}>
          <Image className="q_ans_photos" src={photo.url} thumbnail onClick={()=> setLarge({showed: true, index: i})}/>
          <Modal
          show={isEnlarge.showed} centered size='large'
          onHide={() => setLarge({showed: false, index: i})}>
          <Modal.Header closeButton></Modal.Header>
          <Modal.Body>
            <Image src={props.photos[isEnlarge.index].url} thumbnail/>
          </Modal.Body>
          </Modal>
        </div>
      )}
    </div>
  );
};

export default AnswerPhotos;