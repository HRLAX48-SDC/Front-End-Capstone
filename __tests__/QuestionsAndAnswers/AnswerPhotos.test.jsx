import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import AnswerPhotos from '../../client/src/components/questionsandanswers/AnswerPhotos.jsx';
Enzyme.configure({ adapter: new Adapter()});

const images = [
  {
    id: 4699586,
    url: 'http://res.cloudinary.com/dtnikmimx/image/upload/v1641450048/t8kuusgqlzx4x0qpigrv.jpg'
  },
  {
    id: 4699587,
    url: 'http://res.cloudinary.com/dtnikmimx/image/upload/v1641450048/t8kuusgqlzx4x0qpigrv.jpg'
  }
]
describe('<AnswerPhotos />', () => {
  test('render correctly', () => {
    const preview = renderer.create(<AnswerPhotos photos={images} />).toJSON();
    expect(preview).toMatchSnapshot();
  });
  test('open modal when click photo', () => {
    const photos = shallow(<AnswerPhotos photos={images} />);
    expect(photos.find('Modal').at(0).prop('show')).toBe(false);
    photos.find('.q_ans_photos').at(0).simulate('click');
    expect(photos.find('Modal').at(0).prop('show')).toBe(true);
    expect(photos.find('Modal').at(1).prop('onHide')).toBeDefined();
  });
});