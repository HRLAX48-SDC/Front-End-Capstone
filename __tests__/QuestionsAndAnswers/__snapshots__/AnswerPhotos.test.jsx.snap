// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnswerPhotos /> render correctly 1`] = `
<div
  className="a_photos"
>
  <div>
    <img
      className="q_ans_photos img-thumbnail"
      onClick={[Function]}
      src="http://res.cloudinary.com/dtnikmimx/image/upload/v1641450048/t8kuusgqlzx4x0qpigrv.jpg"
    />
  </div>
  <div>
    <img
      className="q_ans_photos img-thumbnail"
      onClick={[Function]}
      src="http://res.cloudinary.com/dtnikmimx/image/upload/v1641450048/t8kuusgqlzx4x0qpigrv.jpg"
    />
  </div>
</div>
`;
